osfclient/cli.py

```diff
@@ -66,6 +69,37 @@ def _setup_osf(args):
     return OSF(username=username, password=password)
 
 
+def init(args):
+    """Initialize or edit an existing .osfcli.config file
+
+    """
+    # reading existing config file, convert to configparser object
+    config_ = configparser.ConfigParser()
+    config_.read_dict({'osf': config_from_file()})
+
+    if not config_.has_option('osf', 'username'):
+        config_.set('osf', 'username', '')
+    if not config_.has_option('osf', 'project'):
+        config_.set('osf', 'project', '')
```

Comment:
and just to add even more options: you could also use
```python
config['osf'].get('name', default_value_here)
```
without the need for `fallback=...`.